### PR TITLE
Incorporate 'flake8' linting via Travis CI

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

**Comment by frewsxcv** [comment]
If this PR is desired, Travis will need to be enabled for this repository

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+language: python
+python:
+  - 3.3
+  - 3.4
+  - 3.5
+  - pypy3
+install:
+  - pip install flake8
+script:
+  - flake8 --ignore E501 homu
```
Comment (frewsxcv) [inline .travis.yml]: For now at least, I've disabled warnings related to line lengths being >80 characters

Comment (sigmavirus24) [inline .travis.yml]: You could also make it such that the longest line currently is what you set for max-line-length.

**Comment by barosl** [comment]
I have a mixed feeling about PEP 8... I *really* like to keep things one-line if that makes sense, and somehow cannot help feeling the urge to write something like `a*b + c*d`. And I *hate* the 80-width limit. These all are prohibited by PEP 8, and I didn't particularly like that. It is true that, like you did, some warnings can be turned off case by case, but I think that also lowers the advantages of introducing a style enforcement in the first place. It should be applied all-or-nothing, I think.

But at the same time, it is true that this greatly helps multiple people collaborating on the same code base. It also seems to be wanted often, as there was already a PR that suggested the PEP 8 linting. But it was forgotten and silently closed... I felt bad about the author at the time.

**Comment by frewsxcv** [comment]
> urge to write something like `a*b + c*d`

This is a bad practice because one should be grouping things visually via parens, otherwise someone could write something like `a+b * c+d` which is misleading

>  but I think that also lowers the advantages of introducing a style enforcement in the first place

Not really, no. Flake8 is more than style (e.g. catches unused imports, undefined variables, [etc](https://flake8.readthedocs.org/en/2.4.1/warnings.html).) which is very important. 

>  It should be applied all-or-nothing, I think.

Why do you feel this way?

**Comment by barosl** [comment]
> This is a bad practice because one should be grouping things visually via parens

This is a valid concern, I think.

> Flake8 is more than style (e.g. catches unused imports, undefined variables, etc.)

That's interesting! I didn't know it even catchs the undefined variables. I definitely need to read the manual some day.

> Why do you feel this way?

I think using different options means using a slightly different version of PEP 8, leading to multiple style standards out there that are largely compatible, but still slightly incompatible. So if I'm to adhere to PEP 8, I think I should follow all of its rules. The problem is, like I said above, it is really hard for me to like some rules defined in PEP 8. :cry: I think this "all-or-nothing" obsession is my mental problem that needs to be addressed some day. Maybe for now.

**Comment by sigmavirus24** [comment]
So using a configuration file (e.g., tox.ini, setup.cfg, .flake8) and storing the exceptions or modifications can become your style guide. Each OpenStack project has it's own style guide enumerated in their `tox.ini` files. You can configure Flake8 like this and that means that anyone developing the project and using flake8 will not have to run it like Travis does. They can just run `flake8` and it'll just magically work.

As the maintainer of Flake8, I tend to frown on people adding Flake8 to projects without the owner's consent, but the ideology of the tool is to help enforce a Style Guide. That Style Guide can be tweaked, added to, etc. by you the user and we try to be flexible.

If you have questions, I'm happy to help, but I have no strong opinions on whether you start using Flake8 or not. :)

**Comment by cgwalters** [comment]
I wholly agree with the 80 column thing.  It's crazy.  But I'd say flake8/pylint are worth it, let's just go with the config file?

Probably we should land this _after_ merging most of the outstanding PRs so as to not cause lots of merge conflicts.

**Comment by frewsxcv** [comment]
Worth mentioning that https://github.com/barosl/homu/pull/134 also includes flake8 integration.

**Comment by ashcrow** [comment]
I recommend closing this in favor of #134. No conflict in that PR and it includes this as noted.

**Comment by ashcrow** [comment]
@barosl I understand your all-or-nothing thought. I tend to be a little overzealous on code conventions myself and fall [into this bucket](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds) sometimes.

It is possible to [turn off or modify specific errors](http://flake8.readthedocs.org/en/latest/config.html) via `setup.cfg`'s `[flake8]` section. I've found many projects that follow pep8 tend to have at least one exception to it. However, if you'd still like to go all-in on it I'm happy to do some of the grunt work getting code into compliance.

**Comment by frewsxcv** [comment]
@ashcrow @cgwalters: FYI, servo maintains and utilizes their own fork of Homu you can use and submit pull requests to: https://github.com/servo/homu

**Comment by cgwalters** [comment]
@frewsxcv Okay, if there's no motion on https://github.com/barosl/homu/issues/122 then I guess we can resubmit our PRs there for now.

**Pull request closed without merging**